lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -870,6 +871,56 @@ func (c *ContainerProxy) AttachStreams(ctx context.Context, vc *viccontainer.Vic
 	return nil
 }
 
+// Rename() calls the portlayer's containerRenamehandler to update the container name in the handle,
+// and then calls CommitHandler to commit the new name to vSphere
+func (c *ContainerProxy) Rename(vc *viccontainer.VicContainer, newName string) error {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	//retrieve client to portlayer
+	handle, err := c.Handle(vc.ContainerID, vc.Name)
+	if err != nil {
+		return err
+	}
+
+	// Get an API client to the portlayer
+	client := PortLayerClient()
+	if client == nil {
+		err = InternalServerError("wait failed to create a portlayer client")
+		return err
+	}
+
+	// Call the rename functionality in the portlayer.
+	renameParams := containers.NewContainerRenameParamsWithContext(ctx).WithName(newName).WithHandle(handle)
+	result, err := client.Containers.ContainerRename(renameParams)
+	if err != nil {
+		switch err := err.(type) {
+		case *containers.ContainerRenameNotFound:
+			return NotFoundError(vc.Name)
+		case *containers.ContainerRenameConflict:
+			return derr.NewRequestConflictError(fmt.Errorf("container name collision in portlayer"))
+		default:
+			return InternalServerError(err.Error())
+		}
+	}
+
+	h := result.Payload
+
+	// commit handle
+	_, err = client.Containers.Commit(containers.NewCommitParamsWithContext(ctx).WithHandle(h))
```

Comment:
I am not sure this a good fit for this PR however, some work is involved. However, it's a trivial one.